src/node_mgmt/commands.py

```diff
@@ -211,6 +212,61 @@ def readall(s):
     return result
 
 
+def holykell_sensor_address(node):
+    from time import sleep
+    import minimalmodbus
+
+    mod = minimalmodbus.Instrument('/dev/ttyAMA0', 1, debug=True)
+    mod.serial.baudrate = 9600
+    mod.serial.timeout = 3
+    SLAVE_IDS_FOR_HOLYKELL = [7, 8]
+    result = {}
+
+    # confirm that Holykell is accessible on slave id 1
+    logger.info('Checking if communication with holykell is up')
+    try:
+        mod.read_registers(0, 40, 3)
+    except Exception as e:
+        result['Error'] = f'No HPT604 detected on slave 1. Exception: {e}'
+        return result
+
+    # go through targeted slave_ids
+    for slave_id in SLAVE_IDS_FOR_HOLYKELL:
+        try:
```

Comment:
Yeah I agree that this looks a little ugly. I don't immediately have a great suggestion for how to improve it, but I would suggest maybe breaking this up into a few functions. E.g. 
```
try: 
    read_holykell(slave_id)
except minimalmodbus.NoResponseError:
    set_holykell_slave_id(slave_id)
...
```